src/main/java/com/hubspot/smtp/client/SmtpSession.java

```diff
@@ -61,7 +67,7 @@
       SmtpCommand.QUIT,
       SmtpCommand.NOOP);
 
-  private static final Supplier<Executor> SHARED_DEFAULT_EXECUTOR = Suppliers.memoize(SmtpSession::getSharedExecutor);
+  private static final Supplier<Executor> SHARED_DEFAULT_EXECUTOR = () -> Suppliers.memoize(SmtpSession::getSharedExecutor).get();
```

Comment:
I can remove it. I used a `java.util.function.Supplier` while I was working on this, but `Suppliers.memoize` returns a Guava `Supplier`, so this was to convert between the two.